packages/design-system/src/components/slider/slider.tsx

```diff
@@ -0,0 +1,311 @@
+  );
+
+  return (
+    <RACSliderOutput
+      {...rest}
+      ref={ref}
+      className={classNames?.output?.container}
+      style={style}
+    >
+      {children}
+    </RACSliderOutput>
+  );
+});
+
+/**
+ * SliderTrack must be used as a child of Slider and a parent of SliderThumb
+ */
+
+export const SliderTrackContext =
+  createContext<ContextValue<SliderTrackProps, HTMLDivElement>>(null);
+
+export const SliderTrack = forwardRef(function SliderTrack(
+  props: SliderTrackProps,
+  ref: ForwardedRef<HTMLDivElement>,
+) {
+  [props, ref] = useContextProps(props, ref, SliderTrackContext);
+
+  props = useDefaultProps(props, 'SliderTrack');
+
+  const { children: childrenProp, classNames: classNamesProp, ...rest } = props;
+
+  const theme = useTheme();
+
+  const classNames = useMemo(
+    () => mergeClassNames(sliderClassNames, theme.Slider, classNamesProp),
+    [theme.Slider, classNamesProp],
+  );
+
+  const style = useCallback(
+    (renderProps: SliderTrackRenderProps) =>
+      inlineVars(sliderTrackStateVars, renderProps),
+    [],
+  );
+
+  const values = useMemo<
+    [
+      [
+        typeof SliderThumbContext,
+        ContextValue<SliderThumbProps, HTMLDivElement>,
+      ],
+    ]
+  >(() => [[SliderThumbContext, { classNames }]], [classNames]);
+
+  const children = useCallback(
+    (renderProps: SliderTrackRenderProps) => (
+      <Provider values={values}>
+        {callRenderProps(childrenProp, renderProps)}
+      </Provider>
+    ),
+    [childrenProp, values],
+  );
+
+  return (
+    <RACSliderTrack
+      {...rest}
+      ref={ref}
+      className={classNames?.track?.container}
+      style={style}
+    >
+      {children}
+    </RACSliderTrack>
+  );
+});
+
+/**
+ * SliderThumb must be used as a child of SliderTrack
+ */
+
+export const SliderThumbContext =
+  createContext<ContextValue<SliderThumbProps, HTMLDivElement>>(null);
+
+export const SliderThumb = forwardRef(function SliderThumb(
+  props: SliderThumbProps,
+  ref: ForwardedRef<HTMLDivElement>,
+) {
+  [props, ref] = useContextProps(props, ref, SliderThumbContext);
+
+  props = useDefaultProps(props, 'SliderThumb');
+
+  const { children: childrenProp, classNames: classNamesProp, ...rest } = props;
+
+  const theme = useTheme();
+
+  const classNames = useMemo(
+    () => mergeClassNames(sliderClassNames, theme.Slider, classNamesProp),
+    [theme.Slider, classNamesProp],
+  );
+
+  const style = useCallback(
+    (renderProps: SliderThumbRenderProps) =>
+      inlineVars(sliderThumbStateVars, renderProps),
+    [],
+  );
+
+  const values = useMemo<
+    [[typeof AriaLabelContext, ContextValue<LabelProps, HTMLLabelElement>]]
+  >(() => [[AriaLabelContext, {}]], []);
+
+  const children = useCallback(
+    (renderProps: SliderThumbRenderProps) => (
+      <Provider values={values}>
+        <div className={classNames?.thumb?.thumb}>
+          {callRenderProps(childrenProp, {
+            ...renderProps,
+            orientation: 'horizontal',
```

Comment:
Because the RAC supports a vertical orientation, ours should too

Reply:
Since we don't have a design for that, how do you think i should handle the vertical style?